pkg/apis/performance/v1alpha1/performanceprofile_types.go

```diff
@@ -12,7 +12,15 @@ type PerformanceProfileSpec struct {
 	CPU *CPU `json:"cpu,omitempty"`
 	// HugePages defines set of huge pages related parameters.
 	HugePages *HugePages `json:"hugepages,omitempty"`
-	// NodeSelector is a selector which must be true for the performance profile to fit on a node.
+	// MachineConfigLabels defines the labels to add to the MachineConfigs the operator creates. They have to be
+	// used as MachineConfigSelector in the MachineConfigPool which targets this performance profile.
+	MachineConfigLabels map[string]string `json:"machineConfigLabels,omitempty"`
```

Comment:
can we successfully default any of these values. Is there any somewhat sane heuristic we can use to key of one value and in 99% of the cases detect the rest of the values?

for example, keying off node selector. ```node-role.kubernetes.io/worker-rt``` could we default the mcp and mc labels to be ```machineconfiguration.openshift.io/role: worker-rt```. That way people only actually supply the node selector to our api most of the time.

Reply:
I have no clear opinion on this.
Looking the default MCPs, the node selectors and mc selectors match indeed. But they don't have a role label at all.
I fear a bit that it's difficult to educate users how our defaults work. By having to explicitly set everything, they know what they have to do, especially setting a label on the MCP, and how things relate to each other. At least when they read the comments / doc of the CRD fields...

Reply:
just looking at the default mcp, node roles, and machineConfigs. it looks like there is an established naming convention. There's nothing forcing this convention to always be the case, but conventions allow us to make some assumptions when presented with little information.


The MCPs and MCs appear to align with a matching node role. So, given the role ```worker```, you'll find a ```worker``` MCP and ```worker``` MCs.

I'd expect the same convention to be followed with custom roles as well. A ```worker-rt``` node role should have a corresponding custom ```worker-rt``` MCP that matches a super set of both ```worker``` and ```worker-rt``` MachineConfigs.

This `OR` logic of matching MCs for both worker and worker-rt seems like it will be common. For example, this match selector on a MCP matches worker+worker-rt. 

```
  machineConfigSelector:
    matchExpressions:
      - {key: machineconfiguration.openshift.io/role, operator: In, values: [worker,worker-rt]}
```

So, all that said. I *think* we can get away with keying everything off the NodeSelector, and then making the other values optional for more advanced use cases. 

thoughts?

Reply:
ack, will give it a try
and thanks for posting the machineConfigSelector, I forgot that we need to select both worker and worker-rt.... that caused all the problems I had yesterday :slightly_smiling_face:

Reply:
done, see pkg/controller/performanceprofile/components/profile